lib/vec.dl

```diff
@@ -0,0 +1,152 @@
+/*
+ * Functions for working with vectors (see basic vector operations defined in
+ * `ddlog_std.dl`).
+ */
+
+/* Applies closure `f` to each element of the vector. */
+function map(v: Vec<'A>, f: function('A): 'B): Vec<'B> {
+    var res = vec_with_capacity(v.len());
+    for (x in v) {
+        res.push(f(x))
+    };
+    res
+}
+
+/* Applies closure `f` to each element of the vector and concatenates the
+ * resulting vectors, returning a flat vector. */
+function flatmap(v: Vec<'A>, f: function('A): Vec<'B>): Vec<'B> {
+    var res = vec_empty();
+    for (x in v) {
+        res.append(f(x))
+    };
+    res
+}
+
+/* Sorts the vector with a key extraction function, but may not preserve the
+ * order of equal elements. */
+function sort_by(v: mut Vec<'A>, f: function('A): 'B) {
+    vec_sort_by(v, f)
+}
+
+/* Returns the element that gives the minimum value from the specified function.
+ * If several elements are equally minimum, the first element is returned.
+ * If the vector is empty, `None` is returned. */
+function min_by(v: Vec<'A>, f: function('A): 'B): Option<'A> {
```

Comment:
arg_min?

Reply:
`arg_min` is more standard math terminology, but `min_by` is going to be easier to locate for someone who is not familiar with the terminology.  Ok, I guess I'll change to `arg_min`